src/sqlancer/clickhouse/ast/ClickHouseBinaryComparisonOperation.java

```diff
@@ -0,0 +1,187 @@
+package sqlancer.clickhouse.ast;
+
+import ru.yandex.clickhouse.domain.ClickHouseDataType;
+import sqlancer.LikeImplementationHelper;
+import sqlancer.Randomly;
+import sqlancer.visitor.BinaryOperation;
+
+
+public class ClickHouseBinaryComparisonOperation extends ClickHouseExpression
+        implements BinaryOperation<ClickHouseExpression> {
+
+    private final ClickHouseBinaryComparisonOperator operation;
+    private final ClickHouseExpression left;
+    private final ClickHouseExpression right;
+
+    public ClickHouseBinaryComparisonOperation(ClickHouseExpression left, ClickHouseExpression right,
+                                               ClickHouseBinaryComparisonOperator operation) {
+        this.left = left;
+        this.right = right;
+        this.operation = operation;
+    }
+
+    public ClickHouseBinaryComparisonOperator getOperator() {
+        return operation;
+    }
+
+    @Override
+    public ClickHouseExpression getLeft() {
+        return left;
+    }
+
+    @Override
+    public ClickHouseExpression getRight() {
+        return right;
+    }
+
+    @Override
+    public String getOperatorRepresentation() {
+        return operation.getTextRepresentation();
+    }
+
+    @Override
+    public ClickHouseConstant getExpectedValue() {
+        ClickHouseConstant leftExpected = left.getExpectedValue();
+        ClickHouseConstant rightExpected = right.getExpectedValue();
+        if (leftExpected == null || rightExpected == null) {
+            return null;
+        }
+
+        return operation.apply(leftExpected, rightExpected);
+    }
+
+    public static ClickHouseBinaryComparisonOperation create(ClickHouseExpression leftVal, ClickHouseExpression rightVal,
+                                                             ClickHouseBinaryComparisonOperator op) {
+        return new ClickHouseBinaryComparisonOperation(leftVal, rightVal, op);
+    }
+
+    public enum ClickHouseBinaryComparisonOperator {
+        SMALLER("<") {
+            @Override
+            ClickHouseConstant apply(ClickHouseConstant left, ClickHouseConstant right) {
+                return left.applyLess(right);
+            }
+
+        },
+        SMALLER_EQUALS("<=") {
+            @Override
+            ClickHouseConstant apply(ClickHouseConstant left, ClickHouseConstant right) {
+                ClickHouseConstant lessThan = left.applyLess(right);
```

Comment:
I noticed that you started implementing all these `execute`/`apply` methods. Do you plan to also support PQS in a later stage? For NoREC and TLP, it is not necessary to implement these.

Reply:
If I'll have enough time.